lib/galaxy_test/selenium/test_history_panel_collections.py

```diff
@@ -132,6 +132,21 @@ def test_output_collection_states_running(self):
             for job in payload["jobs"]:
                 self.dataset_populator.cancel_job(job["id"])
 
+    @selenium_test
+    def test_collection_job_details(self):
+        input_collection, failed_collection = self._generate_partially_failed_collection_with_input()
+        ok_hid = input_collection["hid"]
+        failed_hid = failed_collection["hid"]
+        self.home()
+
+        ok_collection_element = self.history_panel_wait_for_hid_state(ok_hid, "ok")
+        failed_collection_element = self.history_panel_wait_for_hid_state(failed_hid, "error")
+
+        ok_collection_element.collection_job_details_button.wait_for_and_click()
+        self.screenshot("history_panel_collections_job_details_ok")
+        failed_collection_element.collection_job_details_button.wait_for_and_click()
+        self.screenshot("history_panel_collections_job_details_failed")
```

Comment:
I'm quite confused with this test result...
I was expecting the failed collection to have more information about the error, but apparently, both collections share the same `job_source_id`... is this correct or is it a bug?

![history_panel_collections_job_details_ok](https://user-images.githubusercontent.com/46503462/203288808-b4ef79c2-4cfb-4e5a-a1a1-7b1bbc14e9bd.png)

![history_panel_collections_job_details_failed](https://user-images.githubusercontent.com/46503462/203288804-b8f96995-bbf2-4bc4-bd0d-2ee50c14002d.png)

Reply:
That is not correct, is it possible that the page hasn't rendered yet ? It also still says "__DATA_FETCH__" in the screenshot.

Reply:
The page has been rendered for sure, I have some wait time before taking each screenshot locally and I can see the reload of both when running the test visually. Both collections seem to have the same `job_source_id` if I inspect the payload.

Reply:
This is the API response for the contents of the history:
```json
{
    "stats": {
        "total_matches": 2
    },
    "contents": [
        {
            "id": "529fd61ab1c6cc36",
            "name": "exit_code_from_file on collection 1",
            "history_id": "529fd61ab1c6cc36",
            "hid": 6,
            "history_content_type": "dataset_collection",
            "deleted": false,
            "visible": true,
            "type_id": "dataset_collection-529fd61ab1c6cc36",
            "type": "collection",
            "create_time": "2022-11-22T10:40:47.361203",
            "update_time": "2022-11-22T10:40:48.378440",
            "url": "/api/histories/529fd61ab1c6cc36/contents/dataset_collections/529fd61ab1c6cc36",
            "tags": [],
            "model_class": "HistoryDatasetCollectionAssociation",
            "collection_type": "list",
            "populated_state": "ok",
            "populated_state_message": null,
            "element_count": 4,
            "job_source_id": "adb5f5c93f827949",
            "job_source_type": "ImplicitCollectionJobs",
            "job_state_summary": {
                "all_jobs": 4,
                "new": 0,
                "waiting": 0,
                "running": 0,
                "error": 2,
                "paused": 0,
                "deleted_new": 0,
                "resubmitted": 0,
                "queued": 0,
                "ok": 2,
                "failed": 0,
                "deleted": 0,
                "upload": 0
            },
            "contents_url": "/api/dataset_collections/529fd61ab1c6cc36/contents/529fd61ab1c6cc36",
            "collection_id": "529fd61ab1c6cc36",
            "elements_datatypes": [
                "data"
            ]
        },
        {
            "id": "adb5f5c93f827949",
            "name": "Test Dataset Collection",
            "history_id": "529fd61ab1c6cc36",
            "hid": 1,
            "history_content_type": "dataset_collection",
            "deleted": false,
            "visible": true,
            "type_id": "dataset_collection-adb5f5c93f827949",
            "type": "collection",
            "create_time": "2022-11-22T10:40:07.265526",
            "update_time": "2022-11-22T10:40:43.248358",
            "url": "/api/histories/529fd61ab1c6cc36/contents/dataset_collections/adb5f5c93f827949",
            "tags": [],
            "model_class": "HistoryDatasetCollectionAssociation",
            "collection_type": "list",
            "populated_state": "ok",
            "populated_state_message": null,
            "element_count": 4,
            "job_source_id": "adb5f5c93f827949",
            "job_source_type": "Job",
            "job_state_summary": {
                "all_jobs": 1,
                "new": 0,
                "waiting": 0,
                "running": 0,
                "error": 0,
                "paused": 0,
                "deleted_new": 0,
                "resubmitted": 0,
                "queued": 0,
                "ok": 1,
                "failed": 0,
                "deleted": 0,
                "upload": 0
            },
            "contents_url": "/api/dataset_collections/adb5f5c93f827949/contents/adb5f5c93f827949",
            "collection_id": "adb5f5c93f827949",
            "elements_datatypes": [
                "txt"
            ]
        }
    ]
}
```
If you look closely both have the same `"job_source_id": "adb5f5c93f827949"` but now I see that they have different `job_source_type`.

Reply:
Yep, so one is basically `job: 1` and the other is `implicit_collection_job: 1`